scripts/modules/elastic_stack.py

```diff
@@ -24,8 +29,32 @@ class ApmServer(StackService, Service):
 
     def __init__(self, **options):
         super(ApmServer, self).__init__(**options)
-        default_apm_server_creds = {"username": "apm_server_user", "password": "changeme"}
 
+        default_apm_server_creds = {"username": "apm_server_user", "password": "changeme"}
+        self.managed = False
+
+        # run apm-server managed by elastic-agent
+        if self.options.get("apm_server_managed"):
+            self.managed = True
```

Comment:
this should wait for kibana if apm-it is also providing kibana

Reply:
I am specifying _depends_on_kibana_ https://github.com/elastic/apm-integration-testing/pull/1016/files#diff-df11512a4602ab3dcd2e68f8f85f7462684d54050bfcb70ba7f94b39d53bea8aR606

Reply:
gotcha, then I suggest moving the `depends_on` determination logic into `__init__` , via `self.depends_on`